key/src/main/java/net/kyori/adventure/key/Key.java

```diff
@@ -142,6 +150,79 @@ public interface Key extends Comparable<Key>, Examinable, Namespaced {
     return KeyImpl.COMPARATOR;
   }
 
+  /**
+   * Checks if {@code string} can be parsed into a {@link Key}.
+   *
+   * @param string the input string
+   * @return {@code true} if {@code string} can be parsed into a {@link Key}, {@code false} otherwise
+   * @since 4.12.0
+   */
+  static boolean parseable(final @Nullable String string) {
+    if (string == null) {
+      return false;
+    }
+    final int index = string.indexOf(DEFAULT_SEPARATOR);
+    final String namespace = index >= 1 ? string.substring(0, index) : MINECRAFT_NAMESPACE;
+    final String value = index >= 0 ? string.substring(index + 1) : string;
+    return parseableNamespace(namespace) && parseableValue(value);
+  }
+
+  /**
+   * Checks if {@code value} is a valid namespace.
+   *
+   * @param namespace the string to check
+   * @return {@code true} if {@code value} is a valid namespace, {@code false} otherwise
+   * @since 4.12.0
+   */
+  @VisibleForTesting
+  static boolean parseableNamespace(final @NotNull String namespace) {
+    for (int i = 0, length = namespace.length(); i < length; i++) {
+      if (!allowedInNamespace(namespace.charAt(i))) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Checks if {@code value} is a valid value.
+   *
+   * @param value the string to check
+   * @return {@code true} if {@code value} is a valid value, {@code false} otherwise
+   * @since 4.12.0
+   */
+  @VisibleForTesting
+  static boolean parseableValue(final @NotNull String value) {
+    for (int i = 0, length = value.length(); i < length; i++) {
+      if (!allowedInValue(value.charAt(i))) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
```

Comment:
idk if these two methods should be API?

Reply:
Exposed so that anyone parsing or validating a certain part of a `Key` does not have to implement the logic themselves.